modules/mono/glue/cs_files/Basis.cs

```diff
@@ -446,6 +446,26 @@ public Basis(Quat quat)
             _z = new Vector3(xz - wy, yz + wx, 1.0f - (xx + yy));
         }
 
+        public Basis(Vector3 euler)
+        {
+            real_t c;
+            real_t s;
+
+            c = Mathf.Cos(euler.x);
+            s = Mathf.Sin(euler.x);
+            var xmat = new Basis((real_t)1.0, (real_t)0.0, (real_t)0.0, (real_t)0.0, c, -s, (real_t)0.0, s, c);
```

Comment:
Simple numbers, such as 0 and 1, can be represented exactly with extremely few floating digits. It doesn't matter if `0` or `1` is a float or double. In fact, ints are automatically cast to float or double. So you can change `(real_t)1.0` to `1` and `(real_t)0.0` to `0`

Reply:
Thanks, for these pointers.

I've just been playing with this [tool](https://www.h-schmidt.net/FloatConverter/) and it looks like you are entirely correct.  I was under the mistaken impression that floating point representation of integer values was more broken than it appears to be.

You are also correct in saying that in this case int literals would suffice (and avoid the type casting spam)

Thanks for posting https://github.com/godotengine/godot-docs/issues/1296#issuecomment-390308647 let's see which is the preferred approach.